pkg/collector/corechecks/oracle-dbm/oracle.go

```diff
@@ -200,6 +215,22 @@ func (c *Check) Connect() (*sqlx.DB, error) {
 		}
 	}
 
+	if c.config.AgentSQLTrace.Enabled {
+		db.SetMaxOpenConns(1)
+		_, err := db.Exec("ALTER SESSION SET tracefile_identifier='DDAGENT'")
+		if err != nil {
+			log.Warnf("failed to set tracefile_identifier: %v", err)
+		}
+		setEventsStatement := fmt.Sprintf("BEGIN dbms_monitor.session_trace_enable (binds => %t, waits => %t); END;", c.config.AgentSQLTrace.Binds, c.config.AgentSQLTrace.Waits)
```

Comment:
Could this be potentially a security risk can where someone with permissions to the agent configuration (but not the database) can use this to run arbitrary queries?

Reply:
I concatenated parameters with SQL text the instead of properly passing them through the SQL interface because there's a problem with passing `bool` to stored procedure in go-ora driver. That would need some more time to resolve. I created a separate ticket for that: https://datadoghq.atlassian.net/browse/DBM-2441.

There's nevertheless no risk of SQL injection. If, for example, a malicious agent administrator configures an SQL instead of bool values for `binds` and `waits` the configuration won't parse and the check won't execute:

```    agent_sql_trace:
      enabled: 'select sysdate from dual'
      waits: 'select sysdate from dual'
      binds: true
      traced_runs: 10
```

```2023-05-22 15:50:33 CEST | CORE | ERROR | (pkg/collector/corechecks/loader.go:63 in Load) | core.loader: could not configure check oracle-dbm: failed to build check config: yaml: unmarshal errors:
  line 2: cannot unmarshal !!str `select ...` into bool
  line 5: cannot unmarshal !!str `select ...` into bool
2023-05-22 15:50:33 CEST | CORE | ERROR | (pkg/collector/scheduler.go:201 in getChecks) | Unable to load a check from instance of config 'oracle-dbm': JMX Check Loader: check is not a jmx check, or unable to determine if it's so; Python Check Loader: unable to import module 'oracle-dbm': No module named 'oracle-dbm'; Core Check Loader: Could not configure check oracle-dbm: failed to build check config: yaml: unmarshal errors: 
  line 2: cannot unmarshal !!str `select ...` into bool
  line 5: cannot unmarshal !!str `select ...` into bool
2023-05-22 15:50:33 CEST | CORE | ERROR | (pkg/collector/scheduler.go:248 in GetChecksFromConfigs) | Unable to load the check: unable to load any check from config 'oracle-dbm'
```

Even with direct database connections a malicious agent administrator cannot run any queries through `dbms_monitor.session_trace_enable` because of type checking:

'''SQL> BEGIN dbms_monitor.session_trace_enable (binds => 'select sysdate from dual'); end;
  2  /
BEGIN dbms_monitor.session_trace_enable (binds => 'select sysdate from dual'); end;
      *
ERROR at line 1:
ORA-06550: line 1, column 7:
PLS-00306: wrong number or types of arguments in call to 'SESSION_TRACE_ENABLE'
ORA-06550: line 1, column 7:
PL/SQL: Statement ignored'''

Reply:
Gotcha, is it worth adding a type assertion and a comment to this effect in the code?

The reason I ask is because if someone changes `AgentSQLTrace`'s structure (which is in a completely different file) and switches the type of 'Waits' to string unaware of the security implications of that change, we're in trouble. Could it be worth doing `t, ok := c.config.AgentSQLTrace.Binds.(true)` and checking `ok == true`?

Reply:
I see, thanks for the explanation. Improved now.